### PR TITLE
MAYA-121882 Share the cached HdVP2TextureInfo between all materials so that isSRGB

### PR DESCRIPTION
**Pull request by williamkrick**
and uvScaleOffset and re-used correctly when hitting a texture in the cache.

### CODE REVIEW

**Comment by williamkrick** [comment]
@csyshing I'm not sure if this impacting the issues with background material loading but it is in a similar enough area I thought you should be aware of the change.

@JGamache-autodesk we need to save the extra information that is HdVP2TextureInfo so that when another material hits the Maya texture cache we get these values right.

**Review by JGamache-autodesk — DISMISSED** [review]
Looks good. Thanks!

**Comment by JGamache-autodesk** [comment]
I strongly suspect this will help with #2142.

**Review by csyshing — DISMISSED** [review]

**Comment by csyshing** [comment]
@williamkrick @JGamache-autodesk I will test against #2142 to see how's it going.

File: lib/mayaUsd/render/vp2RenderDelegate/material.cpp [hunk 1]
```diff
@@ -2580,20 +2592,23 @@ void HdVP2Material::_UpdateLoadedTexture(
     // function on idle to delete the task object.
     _textureLoadingTasks.erase(path);
 
-    // Check the local cache again, do not overwrite if same texture has
-    // been loaded asynchronously
-    if (_textureMap.find(path) != _textureMap.end()) {
-        return;
```
Comment (williamkrick) [inline lib/mayaUsd/render/vp2RenderDelegate/material.cpp:2586]: The early exit here is wrong. It also stops marking the sprim dirty and triggering a refresh.

File: lib/mayaUsd/render/vp2RenderDelegate/material.h [hunk 2]
```diff
@@ -147,10 +159,11 @@ class HdVP2Material final : public HdMaterial
 
     TfToken _surfaceNetworkToken; //!< Generated token to uniquely identify a material network
 
-    HdVP2ShaderUniquePtr _surfaceShader;    //!< VP2 surface shader instance
-    SdfPath              _surfaceShaderId;  //!< Path of the surface shader
-    HdVP2TextureMap      _textureMap;       //!< Textures used by this material
-    TfTokenVector        _requiredPrimvars; //!< primvars required by this material
+    HdVP2ShaderUniquePtr         _surfaceShader;    //!< VP2 surface shader instance
+    SdfPath                      _surfaceShaderId;  //!< Path of the surface shader
+    static HdVP2GlobalTextureMap _globalTextureMap; //!< Texture in use by all materials in MayaUSD
```
Comment (williamkrick) [inline lib/mayaUsd/render/vp2RenderDelegate/material.h:164]: With a single global static structure we'd never delete the textures until the plugin unloads or Maya exits. At that time it is often not safe to try to destroy the texture, and textures were held in the cache beyond `file -new`.

This way we get the deletion at the right time.

**Review by JGamache-autodesk — DISMISSED** [review]
Very good.

**Review by JGamache-autodesk — APPROVED** [review]

**Pull request merged**